### PR TITLE
CXF-83186 Integrating AWS cloud provider with fabric examples

### PR DESCRIPTION
**Pull request by srushti-patl**
Added AWS provider resources to create VIF in the following examples:
1. port2aws
2. cloudrouter2aws 
3. metal-fabric-2aws

### CODE REVIEW

File: examples/metal-2-fabric-aws-connection/main.tf [hunk 1]
```diff
@@ -2,6 +2,12 @@ provider "equinix" {
   client_id     = var.equinix_client_id
```
Comment (thogarty) [inline examples/metal-2-fabric-aws-connection/main.tf:2]: The example folder should be named `service-token-metal-2-fabric-aws-connection` to stay consistent with example names starting with the submodule they are using for the connection.

File: examples/metal-2-fabric-aws-connection/outputs.tf [hunk 2]
```diff
@@ -1,7 +1,16 @@
 output "metal-connection" {
   value = equinix_metal_connection.metal-connection.id
 }
-
 output "fabric-connection" {
   value = module.metal-2-fabric-connection.primary_connection_id
 }
+output "asw_vpc_id" {
```
Comment (thogarty) [inline examples/metal-2-fabric-aws-connection/outputs.tf:7]: Typo; should be `aws_vpc_id`.

File: examples/metal-2-fabric-aws-connection/terraform.tfvars.example [hunk 3]
```diff
@@ -1,5 +1,6 @@
 equinix_client_id      = "MyEquinixClientId"
 equinix_client_secret  = "MyEquinixSecret"
+equinix_endpoint       = "endpoint_URL"
```
Comment (thogarty) [inline examples/metal-2-fabric-aws-connection/terraform.tfvars.example:3]: I don't think we need to add this for the customer. They can use this to switch to the sandbox environment but if they aren't interested in this then they may think that they need to set this when it already has a default.

File: examples/metal-2-fabric-aws-connection/versions.tf [hunk 4]
```diff
@@ -5,5 +5,9 @@ terraform {
       source  = "equinix/equinix"
       version = ">= 1.20.0"
```
Comment (thogarty) [inline examples/metal-2-fabric-aws-connection/versions.tf:6]: All the equinix versions need to be updated to leverage the bug fix that was made to ensure the AWS provider could be integrated into the examples. I believe it's 1.25.1.

File: examples/port-2-aws-connection/main.tf [hunk 5]
```diff
@@ -1,9 +1,14 @@
 provider "equinix" {
   client_id     = var.equinix_client_id
   client_secret = var.equinix_client_secret
+  endpoint      = var.equinix_endpoint
```
Comment (thogarty) [inline examples/port-2-aws-connection/main.tf:4]: Same comment about endpoint as above.

File: examples/port-2-aws-connection/main.tf [hunk 6]
```diff
@@ -27,3 +32,42 @@ module "create_port_2_aws_connection" {
...
+resource "aws_vpc" "example" {
+  depends_on = [
+    module.port_2_aws_connection
+  ]
+  cidr_block = var.aws_vpc_cidr_block
+}
+
+resource "aws_vpn_gateway" "example" {
+  depends_on = [
+#    aws_vpc.example,
```
Comment (thogarty) [inline examples/port-2-aws-connection/main.tf:51]: Hanging comment that is unneeded. Should be removed.

File: examples/port-2-aws-connection/outputs.tf [hunk 7]
```diff
@@ -1,3 +1,12 @@
 output "aws_connection_id" {
-  value = module.create_port_2_aws_connection.primary_connection_id
+  value = module.port_2_aws_connection.primary_connection_id
+}
+output "asw_vpc_id" {
```
Comment (thogarty) [inline examples/port-2-aws-connection/outputs.tf:4]: Typo in output name for aws.

File: examples/port-2-aws-connection/terraform.tfvars.example [hunk 8]
```diff
@@ -1,5 +1,6 @@
 equinix_client_id      = "MyEquinixClientId"
 equinix_client_secret  = "MyEquinixSecret"
+equinix_endpoint       = "endpoint_URL"
```
Comment (thogarty) [inline examples/port-2-aws-connection/terraform.tfvars.example:3]: Same comment about endpoint in example as above.

File: examples/port-2-aws-connection/versions.tf [hunk 9]
```diff
@@ -5,5 +5,9 @@ terraform {
       source  = "equinix/equinix"
       version = ">= 1.20.0"
```
Comment (thogarty) [inline examples/port-2-aws-connection/versions.tf:6]: Versions need to be updated to be 1.25.1 to include the bug fix for aws integration.

File: examples/cloud-router-2-aws-connection/main.tf [hunk 10]
```diff
@@ -1,8 +1,13 @@
 provider "equinix" {
   client_id     = var.equinix_client_id
   client_secret = var.equinix_client_secret
+  endpoint      = var.equinix_endpoint
```
Comment (thogarty) [inline examples/cloud-router-2-aws-connection/main.tf:4]: Endpoint shouldn't be included for the customer. They will almost always use the default endpoint.

File: examples/cloud-router-2-aws-connection/outputs.tf [hunk 11]
```diff
@@ -1,3 +1,13 @@
 output "module_output" {
   value = module.cloud_router_aws_connection.primary_connection_id
 }
+output "asw_vpc_id" {
```
Comment (thogarty) [inline examples/cloud-router-2-aws-connection/outputs.tf:4]: Typo in output name.

File: examples/cloud-router-2-aws-connection/terraform.tfvars.example [hunk 12]
```diff
@@ -1,5 +1,6 @@
 equinix_client_id      = "<MyEquinixClientId>"
 equinix_client_secret  = "<MyEquinixSecret>"
+equinix_endpoint       = "endpoint_URL"
```
Comment (thogarty) [inline examples/cloud-router-2-aws-connection/terraform.tfvars.example:3]: Don't need the endpoint_url for the customer in the examples as noted in other comments.

File: examples/cloud-router-2-aws-connection/terraform.tfvars.example [hunk 13]
```diff
@@ -19,3 +20,12 @@ additional_info = [
...
 ]
+
+aws_vpc_cidr_block       = "10.255.255.0/28"
+aws_vif_name             = "port2aws"
+aws_vif_vlan             = "320"
+aws_vif_address_family   = "ipv4"
+aws_vif_bgp_asn          = 64999
+aws_vif_amazon_address   = "169.254.0.1/30"
+aws_vif_customer_address = "169.254.0.2/30"
+aws_vif_bgp_auth_key     = "secret"
```
Comment (thogarty) [inline examples/cloud-router-2-aws-connection/terraform.tfvars.example:31]: References to variables that will need to be inserted as secrets should have the `<>` ends.

File: examples/cloud-router-2-aws-connection/versions.tf [hunk 14]
```diff
@@ -5,5 +5,9 @@ terraform {
       source  = "equinix/equinix"
       version = ">= 1.20.0"
```
Comment (thogarty) [inline examples/cloud-router-2-aws-connection/versions.tf:6]: Versions need to be 1.25.1 to include the bug fix for the aws provider integration.

**Review by thogarty — CHANGES_REQUESTED** [review]
Changes requested.

File: examples/cloud-router-2-aws-connection/terraform.tfvars.example (hunk 13 above)
Comment (srushti-patl) [reply examples/cloud-router-2-aws-connection/terraform.tfvars.example:31]: value of aws_vif_bgp_auth_key is secret

**Review by thogarty — APPROVED** [review]
LGTM. Merging.

**Pull request merged**